kafka_consumer/datadog_checks/kafka_consumer/kafka_consumer.py

```diff
@@ -16,12 +27,448 @@ class KafkaCheck(AgentCheck):
+                    partitions = self._kafka_client._client.cluster.partitions_for_topic(topic)
+                topic_partitions.extend([TopicPartition(topic, p) for p in partitions])
+        single_group_offsets_future = self._kafka_client._list_consumer_group_offsets_send_request(
+            group_id=consumer_group, group_coordinator_id=coordinator_id, partitions=topic_partitions
+        )
+        single_group_offsets_future.add_callback(self._single_group_offsets_callback, consumer_group)
+        self._consumer_futures.append(single_group_offsets_future)
+
+    def _single_group_offsets_callback(self, consumer_group, response):
+        """Callback that parses an OffsetFetchResponse and saves it to the consumer_offsets dict.
+
+        consumer_group must be manually passed in because it is not present in the response, but we need it in order to
+        associate these offsets to the proper consumer group.
+        """
+        single_group_offsets = self._kafka_client._list_consumer_group_offsets_process_response(response)
+        for (topic, partition), (offset, _metadata) in single_group_offsets.items():
+            # If the OffsetFetchRequest explicitly specified partitions, the offset could returned as -1, meaning there
+            # is no recorded offset for that partition... for example, if the partition doesn't exist in the cluster.
+            # So ignore it.
+            if offset == -1:
+                self._kafka_client._client.cluster.request_update()  # force metadata update on next poll()
+                continue
+            key = (consumer_group, topic, partition)
+            self._consumer_offsets[key] = offset
+
+    # TODO since this is used to validate the config interface, ideally this would be shared between new and legacy
+    # versions of the check to make sure the interface they accept doesn't diverge if someone updates one but forgets
+    # to update the other copy.
+    def _validate_listed_consumer_groups(self):
+        """Validate the user-specified list of consumer groups."""
+        # _consumer_groups example: {'consumer_group': {'topic': [0, 1]}}
+        # TODO can I allow consumer_groups to be a list rather than requiring a dict?
+        assert isinstance(self._consumer_groups, dict)
+        for consumer_group, topics in self._consumer_groups.items():
+            assert isinstance(consumer_group, string_types)
+            assert isinstance(topics, dict) or topics is None  # topics are optional
+            if topics is not None:
+                for topic, partitions in topics.items():
+                    assert isinstance(topic, string_types)
+                    assert isinstance(partitions, (list, tuple)) or partitions is None  # partitions are optional
+                    if partitions is not None:
+                        for partition in partitions:
+                            assert isinstance(partition, int)
```

Comment:
Let's change all these to `if ...: raise ...` or just only check if root type is correct

Reply:
This is copied from the legacy check, and while I have no problems with dropping the asserts, I also don't think it's worth the effort... so could you guys fix it in the legacy check to whatever level of validation and error messages you want and then I'll forklift whatever you do over into the new check?

Reply:
You can simply remove these asserts, that's fine imo

Reply:
Discussed on slack, agreed that leaving the asserts--though bad practice for prod code--is still better than not having them as it makes it clearer to users what the expected interface is and cuts down on support tickets...